### DISCUSSION TITLE
[V5] Custome Rules Validation

### DISCUSSION THREAD

**Discussion by Pratikkshirsagar** [adonisjs/core#1692] [Help] [answered]
Hi,
I am trying to set a default name length value without passing the parameter in the in schema validator but it throw the error of  ```Cannot read property 'maxLength' of undefined```

```contracts/validator.ts ```
```ts 
  export interface Rules {
    name(options?: { maxLength: number }): Rule;
  }
}
```

```start/validationRules.ts ```

```ts 
validator.rule('name', (value, [{ maxLength = 50 }], { pointer, arrayExpressionPointer, errorReporter }) => {
  const regex = /^[A-Za-z\s]{1,}[\']{0,1}[A-Za-z\s]{0,}$/;
  const name = regex.test(value);

  if (!name || value.trim().length >= maxLength) {
    errorReporter.report(pointer, 'name', 'Invalid name', arrayExpressionPointer);
  }
});
```

```app/validators/UserValidator ```
```ts 
export default class UserValidator {
  constructor(public data: Record<string, unknown>) {}

  public schema = schema.create({
    firstName: schema.string({}, [rules.name()]),
    middleName: schema.string.optional({}, [rules.name()]),
    lastName: schema.string({}, [rules.name()]),
  });
}
```
Is there any way to set the default parameter

**Comment by thetutlage** [comment, marked as answer]
You get the error because you are trying to destructure an object that does exists. Changing the code to following should work fine

```ts
validator.rule('name', (value, [options], { pointer, arrayExpressionPointer, errorReporter }) => {
  const maxLength = options?.maxLength || 50

  const regex = /^[A-Za-z\s]{1,}[\']{0,1}[A-Za-z\s]{0,}$/;
  const name = regex.test(value);

  if (!name || value.trim().length >= maxLength) {
    errorReporter.report(pointer, 'name', 'Invalid name', arrayExpressionPointer);
  }
});
```

**Comment by Lakshamana** [comment]
Not to necrobump this issue, but it feels that in adonisjs v5 custom validation subject doesn't appear to have a dedicated page on docs. Maybe I'm missing something, but a naive search on search bar won't lead to any relevant result
![image](https://user-images.githubusercontent.com/29443505/129617669-4addd9d8-b70f-4e0b-96e8-cf41076a9d7e.png)
